client/blocks/sharing-preview-pane/index.jsx

```diff
@@ -140,6 +109,36 @@ class SharingPreviewPane extends PureComponent {
 			siteName,
 		};
 
+		/**
+		 * Props to pass to the preview component. Will be populated with the connection
+		 * specific data if the selected service is connected.
+		 *
+		 * @type {Object}
+		 */
+		const previewProps = { ...postProps };
+
+		const connection = find( connections, { service: selectedService } );
+
+		if ( ! connection ) {
+			previewProps.hidePostPreview = true;
```

Comment:
This is something that I don't like about `const`, you can still modify the object.

We could probably move things about a bit to avoid some of that, and maybe a bit of object copying. I can't make a suggestion across the right bit of code but I think this might be equivalent to lines 98-138

```js
		const connection = find( connections, { service: selectedService } );

		/**
		 * Props to pass to the preview component. Will be populated with the connection
		 * specific data if the selected service is connected.
		 *
		 * @type {Object}
		 */
		const previewProps = {
			articleUrl,
			articleTitle,
			articleContent,
			articleSummary,
			message,
			imageUrl,
			seoTitle,
			siteDomain,
			siteIcon,
			siteName,
			hidePostPreview: ! connection,
			...(
				connection
				? {
					externalDisplay: connection.external_display,
					externalName: connection.external_name,
					externalProfileURL: connection.external_profile_URL,
					externalProfilePicture: connection.external_profile_picture,
				}
				: {}
			),
		};
```

We could also rely on there having to be an ID property of the connection and simplify this to:

```js
		const connection = find( connections, { service: selectedService } ) ?? {};

		/**
		 * Props to pass to the preview component. Will be populated with the connection
		 * specific data if the selected service is connected.
		 *
		 * @type {Object}
		 */
		const previewProps = {
			articleUrl,
			articleTitle,
			articleContent,
			articleSummary,
			message,
			imageUrl,
			seoTitle,
			siteDomain,
			siteIcon,
			siteName,
			hidePostPreview: ! connection.ID,
			externalDisplay: connection.external_display,
			externalName: connection.external_name,
			externalProfileURL: connection.external_profile_URL,
			externalProfilePicture: connection.external_profile_picture,
		};
```

That would mean that we're always adding these undefined properties, but my guess is that might be more permformant than creating an intermediate object, and removes all the conditional stuff.